mlx/backend/metal/compiled.cpp

```diff
@@ -38,8 +41,8 @@ inline void build_kernel(
   int cnt = 0;
 
   // Start the kernel
-  os << "[[host_name(\"" << kernel_name << "\")]]" << std::endl
-     << "[[kernel]] void " << kernel_name << "(" << std::endl;
+  os << "[[host_name(\"" << kernel_name << "\")]]\n"
+     << "[[kernel]] void " << kernel_name << "(\n";
```

Comment:
Do the explicit newlines make it easier to follow/smaller?

Reply:
For ostringstream `std::endl` gets resolved to `\n`. My only rationale here is that I think `os << "a" << "\n"` is a lot slower (in the worst case) than `os << "a\n"`.

I was thinking to change all of it to string concatenation per our previous conversation.. but I didn't want to blow up this diff. It probably has a small to negligible impact on startup cost but honestly I haven't measured it yet so 🤷‍♂️ .